models/articles.models.js

```diff
@@ -1,14 +1,22 @@
 const db = require("../db/connection.js");
 
 async function fetchArticleById(article_id) {
-    //const {rows} =await db.query(`SELECT article_id FROM articles`)
-    //const validIds = rows.map((row) => row.article_id)
-    console.log('model')
     if (!Number(article_id)) return Promise.reject({ status: 400, message: 'Bad request' })
     return db.query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id]).then((article) => {
         if (!article.rows.length) return Promise.reject({ status: 404, message: 'Article id not found' })
         return article.rows[0];
     });
 }
 
-module.exports = { fetchArticleById };
\ No newline at end of file
+async function fetchCommentsOnArticle(article_id){
+    if (!Number(article_id)) return Promise.reject({ status: 400, message: 'Bad request' })
+    const validArticleIDs = await db.query(`SELECT article_id FROM articles;`).then((articles) => {
+        return articles.rows.map((article) => article.article_id);
+    });
+    if (!validArticleIDs.includes(Number(article_id))) return Promise.reject({ status: 404, message: 'Article id not found' })
+    return db.query(`SELECT * FROM comments WHERE article_id = ${article_id} ORDER BY created_at DESC;`).then((comments) => {
```

Comment:
Aaahh! You are opening yourself up to SQL injection here! 😱 Please fix!

Reply:
Hey Reyhaan, I thought I'd avoided SQL injection by having a validArticleIDs array above, i.e. line 17 is only reached if it is a valid article ID, let me know if i've misunderstood :)

Reply:
Yes, you are right! I will put my pitchfork down! 😅, however, lets stick to convention and use ```$1```

Reply:
hahaha okay no worries, will do